WithXXX functions to configure the Authenticator options cause a type inteference problem.

The code from the example Go app initialises the authenticator like this:

```
authN, err := authentication.New(ctx, zitadel.New(*domain), *key,
	openid.DefaultAuthentication(*clientID, *redirectURI, *key),
)
```

which works fine, but if you try to pass in an option with the WithXXX functions, the Go compiler (1.25) can't infer the types correctly. e.g. 

```
authN, err := authentication.New(ctx, zitadel.New(*zitadelDomain), key,
	openid.DefaultAuthentication(*zitadelClientID, *zitadelRedirectURI, key),
	authentication.WithPostLogoutRedirectURI[authentication.Ctx](*zitadelPostLogoutRedirectURI),
)
```

generates the following compiler error:

```
main.go:131:3: in call to authentication.New, type authentication.Option[authentication.Ctx] of authentication.WithPostLogoutRedirectURI[authentication.Ctx](*zitadelPostLogoutRedirectURI) does not match inferred type authentication.Option[*"github.com/zitadel/zitadel-go/v3/pkg/authentication/oidc".UserInfoContext[*"github.com/zitadel/oidc/v3/pkg/oidc".IDTokenClaims, *"github.com/zitadel/oidc/v3/pkg/oidc".UserInfo]] for authentication.Option[T]
```

I'm still fairly new to Go, but I think the problem is the return value of DefaultAuthentication has a pointer for its generic type:

`authentication.HandlerInitializer[*UserInfoContext[*oidc.IDTokenClaims, *oidc.UserInfo]]`

which doesn't match `Option[T Ctx]`

--- a/pkg/authentication/oidc/authenticate.go
+++ b/pkg/authentication/oidc/authenticate.go
@@ -14,6 +14,8 @@ import (
 	"github.com/zitadel/zitadel-go/v3/pkg/zitadel"
 )
 
+type DefaultContext = UserInfoContext[*oidc.IDTokenClaims, *oidc.UserInfo]
+
 type Ctx[C oidc.IDClaims, S rp.SubjectGetter] interface {
 	authentication.Ctx
 	New() Ctx[C, S]
@@ -64,11 +66,11 @@ func ClientIDSecretAuthentication(clientID, clientSecret, redirectURI string, sc
 // DefaultAuthentication is a short version of [WithCodeFlow[*UserInfoContext[*oidc.IDTokenClaims, *oidc.UserInfo], *oidc.IDTokenClaims, *oidc.UserInfo]]
 // with the client_id, redirectURI and encryptionKey and optional scopes.
 // If no scopes are provided, `"openid", "profile", "email"` will be used.
-func DefaultAuthentication(clientID, redirectURI string, key string, scopes ...string) authentication.HandlerInitializer[*UserInfoContext[*oidc.IDTokenClaims, *oidc.UserInfo]] {
+func DefaultAuthentication(clientID, redirectURI string, key string, scopes ...string) authentication.HandlerInitializer[*DefaultContext] {
 	if len(scopes) == 0 {
 		scopes = []string{oidc.ScopeOpenID, oidc.ScopeProfile, oidc.ScopeEmail}
 	}
-	return WithCodeFlow[*UserInfoContext[*oidc.IDTokenClaims, *oidc.UserInfo], *oidc.IDTokenClaims, *oidc.UserInfo](
+	return WithCodeFlow[*DefaultContext, *oidc.IDTokenClaims, *oidc.UserInfo](
 		PKCEAuthentication(clientID, redirectURI, scopes, httphelper.NewCookieHandler([]byte(key), []byte(key))),
 	)
 }

